src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -10,18 +10,19 @@ export class StacheNavService {
     private windowRef: StacheWindowRef) { }
 
   public navigate(route: any): void {
-    let extras: any = {};
-    const currentPath = this.router.url.split('#')[0].substring(1);
+    let extras: any = { queryParamsHandling: 'merge'};
+    const currentPath = this.router.url.split('?')[0].split('#')[0].substring(1);
+
     if (this.isExternal(route)) {
       this.windowRef.nativeWindow.location.href = route.path;
       return;
     }
 
     if (route.fragment) {
-      const element = this.windowRef.nativeWindow.document.getElementById(route.fragment);
+      const element = this.windowRef.nativeWindow.document.querySelector(`#${route.fragment}`);
```

Comment:
Curious why you went with `querySelector`? It's a bit slower than `getElementById`.

Reply:
I went with `querySelector` because most examples I have seen were posted using it.  It seemed like a more standard practice.